pkg/vm/stack.go

```diff
@@ -83,35 +83,16 @@ func (e *Element) BigInt() *big.Int {
 // TryBool attempts to get the underlying value of the element as a boolean.
 // Returns error if can't convert value to boolean type.
 func (e *Element) TryBool() (bool, error) {
-	switch t := e.value.(type) {
-	case *BigIntegerItem:
-		return t.value.Int64() != 0, nil
-	case *BoolItem:
-		return t.value, nil
-	case *ArrayItem, *StructItem:
-		return true, nil
-	case *ByteArrayItem:
-		for _, b := range t.value {
-			if b != 0 {
-				return true, nil
-			}
-		}
-		return false, nil
-	case *InteropItem:
-		return t.value != nil, nil
-	default:
-		return false, fmt.Errorf("can't convert to bool: " + t.String())
+	if e.value == nil {
+		return false, errors.New("nil value")
 	}
+	return e.value.Bool(), nil
```

Comment:
It makes `TryBool` useless. If anything can be converted to `Boolean` successfully, then we shouldn't be trying, we should just convert and be done with it.

Reply:
Removed.